### ISSUE TITLE
Anyone already working on including this in transformers? 

### ISSUE THREAD

**Issue by l4b4r4b4b4** [ironjr/grokfast#2]
Ill try my best, but thought to check if there is anyone else wanting to try this in context of transformers trainer.

**Comment by ironjr** [comment]
Although it is a fairly small one, the main algorithmic data experiments are done with a 2-layer Transformer with 400k parameters. I'll leave this issue open so people can share their thoughts.

**Comment by l4b4r4b4b4** [comment]
worked the approach into a fork of HF transformer trainer. 
Runs without errors using trl ORPO trainer and unsloth. Will do some testing and report over the coming week. 
Since I did not follow a single best practice from the transformer library no PR yet but for anyone who wants to try it out: https://github.com/l4b4r4b4b4/transformers/blob/main/src/transformers/trainer.py

**Comment by phalexo** [comment]
Is there the expectation that all weights within a network are trainable or can it be used for fine-tuning when only some layers are trainable, @ironjr ? 

I tried to insert the code into the Trainer inner training step as well but I get an error about NoneType.

**Comment by lucasjinreal** [comment]
@l4b4r4b4b4 hows the result going

**Comment by HydrogenBombaklot** [comment]
@l4b4r4b4b4 any update?

**Comment by l4b4r4b4b4** [comment]
Had everything implemented in a transformers fork. I think @ehartford https://github.com/cognitivecomputations/grokadamw took it a bit further.

**Comment by ehartford** [comment]
I got creative, my implementation is inspired by the paper rather than a direct implementation of it

**Comment by phalexo** [comment]
Has it been tried on larger models to assess training time reduction? I am
planning to deploy llama 3.1 70b instruct for machine translation and
wondering if fine-tuning of it could benefit.

**Comment by ehartford** [comment]
Mine so far I'm seeing only marginal divergence vs adamw-fused

An improvement but not an obvious slam dunk.

Maybe I need to improve the default grokking functions

![image-8.png](https://github.com/user-attachments/assets/397c77a3-09a2-45a1-a610-2b9c2f88cfb0)

**Comment by phalexo** [comment]
I am looking at the graphic and the two loss lines seem to track each
other, there is no divergence, The absolute difference is likely just the
artefact of
how the two are initialized.

**Comment by ehartford** [comment]
Well, I suppose you know best.

Here is how they started.

<img width="1468" alt="image" src="https://github.com/user-attachments/assets/e4dbefbd-8593-46d9-bf0c-75fc0e6765c9">

**Comment by BradKML** [comment]
Are there open data models that can test the efficiency of GrokFast? Only GPT-NeoX and the other one came to mind atm https://github.com/EleutherAI/gpt-neox https://github.com/openlm-research/open_llama
